### ISSUE TITLE
App or Site in DoubleClick

### ISSUE THREAD

**Issue by marcelomalcher** [google/openrtb-doubleclick#59]
Hi Osvaldo, 

First, I would like to thank you for that lib. It is being very helpful to work with it to do the mapping between DC and OpenRTB specs.

I am just confused by the way the lib creates an App or a Site. From what I know, OpenRTB states that a request coming from a non-browser application contains an App object and a request from a website contains a Site object. 

While reading from DoubleClick spec, I understand that a request comes from an application when you have a `Mobile` object and `Mobile.is_app = true`, right? 

But looking into the code I can read:

```
    if (dcRequest.hasMobile()) {
      App.Builder app = buildApp(dcRequest);
      if (app != null) {
        request.setApp(app);
      }
    } else {
      Site.Builder site = buildSite(dcRequest);
      if (site != null) {
        request.setSite(site);
      }
    }

```

From the code above, the lib is creating an App object when the `Mobile` object is present in the request, despite if is a request from the mobile web or an in-app request.

So, shouldn't be added this other condition like the code below?

```
    if (dcRequest.hasMobile() && dcRequest.getMobile().getIsApp()) {
      App.Builder app = buildApp(dcRequest);
      ...
    } else {
      Site.Builder site = buildSite(dcRequest);
      ...
    }
```

**Comment by opinali** [comment]
Hi Marcelo, thanks for reporting this! It's a bug, we should definitely consider the `is_app` flag, like in your suggested snippet. Will fix that in the next update.

**Comment by opinali** [comment]
Fixed now with 0.9.8.
